experimental/skills/DialogToDialog/DialogRootBot/AdapterWithErrorHandler.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.BotBuilderSamples.DialogRootBot.Middleware;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
+
+namespace Microsoft.BotBuilderSamples.DialogRootBot
+{
+    public class AdapterWithErrorHandler : BotFrameworkHttpAdapter
+    {
+        public AdapterWithErrorHandler(IConfiguration configuration, ICredentialProvider credentialProvider, AuthenticationConfiguration authConfig, ILogger<AdapterWithErrorHandler> logger, ConversationState conversationState = null)
+            : base(configuration, credentialProvider, authConfig, logger: logger)
+        {
+            OnTurnError = async (turnContext, exception) =>
+            {
+                // Log any leaked exception from the application.
+                logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.Message}");
+
+                // Send a message to the user
+                var errorMessageText = "The bot encountered an error or bug.";
+                var errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.IgnoringInput);
+                await turnContext.SendActivityAsync(errorMessage);
+
+                errorMessageText = "To continue to run this bot, please fix the bot source code.";
+                errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.ExpectingInput);
+                await turnContext.SendActivityAsync(errorMessage);
+
+                if (conversationState != null)
```

Comment:
Not sure if there is any cleanup to do around active skills.

If there is an active skill:
1. Will it time out, or does the skill have to add some logic to make that happen.
2. More importantly, won't the current ConversationIdFactory try to reuse the old host-to-skill conversation ID? (Presumably, the host-to-user conversation ID will be the same.) If we don't cancel the skill and invoke it anew, it may try to continue mid-course, unless the skill has logic to abort and start over if there are input parameters.

Reply:
Resolving this one for now, I added a note to the DialogToDialog items in https://github.com/microsoft/botframework-sdk/issues/5582 so I can address this issue in combination with resetting the dialog stack on the parent